pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -570,31 +570,75 @@ def diagonalize_mcms(tape):
 
         from pennylane.ftqc import diagonalize_mcms, ParametricMidMeasureMP
 
-        dev = qml.device("default.qubit")
+        dev = qml.device("default.qubit", shots=1000)
 
         @diagonalize_mcms
-        @qml.qnode(dev)
+        @qml.qnode(dev, method="one-shot")
         def circuit(x):
-            qml.RY(x[0], wires=0)
-            ParametricMidMeasureMP(0, angle=x[1], plane="XY")
-            return qml.expval(qml.Z(0))
+            qml.RX(x, wires=0)
+            m = measure_y(0)
+            qml.cond(m, qml.X)(1)
+            return qml.expval(qml.Z(1))
 
     Applying the transform inserts the relevant gates before the measurement to allow
     measurements to be in the Z basis, so the original circuit
 
-    >>> print(qml.draw(circuit, level=0)([np.pi/4, np.pi]))
-    0: ──RY(0.79)──┤↗ˣʸ(3.14)├─┤  <Z>
+    >>> print(qml.draw(circuit, level=0)(np.pi/4))
+    0: ──RX(0.79)──┤↗ʸ├────┤
+    1: ─────────────║────X─┤  <Z>
+                    ╚════╝
 
     becomes
 
-    >>> print(qml.draw(circuit)([np.pi/4, np.pi]))
-    ──RY(0.79)──Rϕ(-3.14)──H──┤↗├─┤  <Z>
+    >>> print(qml.draw(circuit)(np.pi/4))
+    0: ──RY(0.79)──S†──H──┤↗├────┤
+    1: ────────────────────║───X─┤  <Z>
+                           ╚═══╝
+
+
+    .. details::
+        :title: Conditional measurements
+
+        The transform can also handle diagonalization of conditional measurements created by
+        :func:`qml.ftqc.cond_meas <pennylane.ftqc.cond_meas>`. This is done by replacing the
+        measurements for the true and false condition with conditional diagonalizing gates,
+        and a single measurement in the computational basis:
+
+        .. code-block:: python3
+
+            from pennylane.ftqc import diagonalize_mcms, measure_x
+
+            dev = qml.device("default.qubit")
+
+            @diagonalize_mcms
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RY(x[0], wires=0)
+                qml.RX(x[1], wires=1)
+                m = qml.measure(0)
+                m2 = cond_meas(m, measure_x, measure_y)(1)
+                qml.cond(m2, qml.X)(1)
+                return qml.expval(qml.Z(1))
+
+        This circuit diagonalizes to:
+
+        >>> print(qml.draw(circuit)([np.pi, np.pi/4]))
+        0: ──RY(3.14)──┤↗├───────────────────┤
+        1: ──RX(0.79)───║───H──S†──H──┤↗├──X─┤  <Z>
```

Comment:
I guess the first `H` in `wire=1` is the diagonalizing gate for `measure_x`, while `S` and `H` are for `measure_y`. I'm wondering if there is an easy way to make it more straightforward to show the `cond_measure` takes `measure_x` and `measure_y`.

Reply:
Yes, that's right! I'll add the original, untransformed circuit before the diagonalized one to provide more context.

Reply:
Actually the drawer becomes utterly confused when trying to handle circuits where an operator is both conditional and a condition for other things, making this able to be drawn would be its own PR. I'll add a little text blurb for context 😅

Reply:
I'm not sure how clear the text I added was, lmk if it seems helpful

Reply:
Thank you Lillian! Looking good to me.